### PR TITLE
Address two bugs with the handling of prefixes

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

While implementing the `delete=clear+save` we saw a number of problems. Here are the ones for prefixes

## Proposal

Two bugs were addressed:
* One when we insert a prefix bug it is already matched by existing prefixes.
* One in `MapView` where a prefix has been deleted and the `for_each_key_*` is used for the same prefix. Then we need to remove the prefix from consideration.

## Test Plan

The CI and a better check.

## Release Plan

None.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: linera-views/src/key_value_store_view.rs [hunk 1]
```diff
@@ -547,9 +548,7 @@ where
         if self.was_cleared {
             return Ok(None);
         }
-        let iter = self.deleted_prefixes.iter();
-        let mut lower_bound = GreatestLowerBoundIterator::new(0, iter);
-        if !lower_bound.is_index_absent(index) {
+        if !is_index_absent(&self.deleted_prefixes, index) {
```
Comment (afck) [inline linera-views/src/key_value_store_view.rs:551]: Maybe it would be better to implement `is_index_present` instead, so we don't have double negations.

**Review by afck — APPROVED** [review]

**Comment by ma2bd** [comment]
From the summary I cannot tell if the bug is serious (corrupted data) or not (aspirational storage invariant)

**Comment by MathieuDutSik** [comment]
> From the summary I cannot tell if the bugs are serious (corrupted data?) or not (aspirational storage invariant?)

Serious.

File: linera-views/src/key_value_store_view.rs (hunk 1 above)
Comment (MathieuDutSik) [reply linera-views/src/key_value_store_view.rs:551]: Possibly, but for another PR.

**Pull request merged**